apps/studio/src/features/editing-experience/components/ScheduledPublishingModal/ScheduledPublishingModal.tsx

```diff
@@ -0,0 +1,427 @@
+            }
+          },
+        )}
+      >
+        <ModalOverlay />
+        <ModalContent>
+          <ModalHeader mr="3.5rem">Publish this page?</ModalHeader>
+          <ModalCloseButton size="lg" />
+          <ModalBody>
+            <FormProvider {...methods}>
+              <VStack align="stretch" spacing="1.5rem">
+                <VStack spacing="1rem" align="stretch">
+                  <VStack align="stretch" spacing="1.5rem">
+                    <Text textStyle="subhead-1">
+                      When should we publish this page?
+                    </Text>
+                    <VStack spacing="0.75rem" align="stretch">
+                      <PublishModeCard
+                        icon={BiRocket}
+                        title="Publish now"
+                        description="Changes will be live on your site in approximately 5-10 minutes."
+                        isSelected={publishMode === PublishMode.NOW}
+                        onSelect={() =>
+                          methods.setValue("publishMode", PublishMode.NOW)
+                        }
+                      />
+                      <PublishModeCard
+                        icon={BiTimeFive}
+                        title="Publish later"
+                        description="Let us know when the page should start publishing."
+                        isSelected={publishMode === PublishMode.SCHEDULED}
+                        onSelect={() =>
+                          methods.setValue("publishMode", PublishMode.SCHEDULED)
+                        }
+                      />
+                    </VStack>
+                  </VStack>
+                  {publishMode === PublishMode.SCHEDULED && (
+                    <SchedulePublishDetails />
+                  )}
+                </VStack>
+                {publishMode === PublishMode.SCHEDULED &&
+                  isScheduledPublishValid && <SchedulePublishBanner />}
+              </VStack>
+            </FormProvider>
+          </ModalBody>
+          <ModalFooter>
+            <Button
+              mr={3}
+              onClick={onClose}
+              variant="clear"
+              color="base.content.strong"
+            >
+              No, don't publish
+            </Button>
+            <Button type="submit" isLoading={isPublishingNow || isScheduling}>
+              {publishMode === PublishMode.NOW
+                ? "Publish now"
+                : "Schedule publish"}
+            </Button>
+          </ModalFooter>
+        </ModalContent>
+      </form>
+    </Modal>
+  )
+}
+
+/**
+ * if the date provided is equal to the earliestSchedule's date, the earliest allowable time should be set to the FIRST
+ * time slot after the current minimum allowable time
+ * @param selectedDate Date selected inside the datepicker
+ * @param earliestSchedule Earliest schedule time, based on the current date and MINIMUM_SCHEDULE_LEAD_TIME_MINUTES
+ * @returns
+ */
+const getEarliestAllowableTime = (
```

Comment:
can move into a `utils` folder in this file and write a simple unit test for this?